concepts/05 UI Components/PivotGrid/035 Use CustomStore/20 Remote Operations.md

```diff
@@ -54,6 +55,396 @@ The example below shows how to implement the **load** function. Note that in thi
+          area: "row"
+        },
+        {
+          caption: "Subcategory",
+          dataField: "ProductSubcategoryName",
+          width: 250,
+          sortBySummaryField: "SalesAmount",
+          sortBySummaryPath: [],
+          sortOrder: "desc",
+          area: "row"
+        },
+        {
+          caption: "Product",
+          dataField: "ProductName",
+          area: "row",
+          sortBySummaryField: "SalesAmount",
+          sortBySummaryPath: [],
+          sortOrder: "desc",
+          width: 250
+        },
+        {
+          caption: "Date",
+          dataField: "DateKey",
+          dataType: "date",
+          area: "column"
+        },
+        {
+          caption: "Amount",
+          dataField: "SalesAmount",
+          summaryType: "sum",
+          format: "currency",
+          area: "data"
+        },
+        {
+          caption: "Store",
+          dataField: "StoreName"
+        },
+        {
+          caption: "Quantity",
+          dataField: "SalesQuantity",
+          summaryType: "sum"
+        },
+        {
+          caption: "Unit Price",
+          dataField: "UnitPrice",
+          format: "currency",
+          summaryType: "sum"
+        },
+        {
+          dataField: "Id",
+          visible: false
+        }
+      ]
+    };
+  }
+}
+
+@NgModule({
+  imports: [BrowserModule, DxPivotGridModule, HttpClientModule],
+  declarations: [AppComponent],
+  bootstrap: [AppComponent]
+})
+export class AppModule {}
+
+platformBrowserDynamic().bootstrapModule(AppModule);
+
+
+    <!--Vue-->
+<template>
+  <DxPivotGrid
+    :data-source="dataSource"
+  />
+</template>
+<script>
+import { DxPivotGrid, DxScrolling } from "devextreme-vue/pivot-grid";
+
+import CustomStore from "devextreme/data/custom_store";
+import "whatwg-fetch";
+
+function isNotEmpty(value) {
+  return value !== undefined && value !== null && value !== "";
+}
+
+export default {
+  components: {
+    DxPivotGrid,
+  },
+  data() {
+    return {
+      dataSource: {
+        remoteOperations: true,
+        store: new CustomStore({
+          key: "OrderNumber",
+          load: function (loadOptions) {
+            let params = "?";
+            [
+              "skip",
+              "take",
+              "requireTotalCount",
+              "requireGroupCount",
+              "sort",
+              "filter",
+              "totalSummary",
+              "group",
+              "groupSummary",
+            ].forEach(function (i) {
+              if (i in loadOptions && isNotEmpty(loadOptions[i])) {
+                params += `${i}=${JSON.stringify(loadOptions[i])}&`;
+              }
+            });
+            params = params.slice(0, -1);
+            return fetch(
+              `https://js.devexpress.com/Demos/WidgetsGalleryDataService/api/Sales/orders${params}`
+            )
+              .then((response) => response.json())
+              .then((result) => {
+                return new Promise((resolve) => {
```

Comment:
Let's handle 'response.ok' in our demo code:

    if (!response.ok) {
      throw new Error('Network response was not ok');
    }

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful

Reply:
Yep. Great idea.
Fixed - https://github.com/DevExpress/devextreme-documentation/pull/2316/commits/d596d84c18a3c9b67b9c9ec86cf0053b5431d0e8